infra/src/dsl/processing/rendering-pipeline.ts

```diff
@@ -36,25 +43,44 @@ export function prepareServicesForEnv<T extends ServiceOutputType>(options: {
 }
 
 /**
- * This is an important function. It converts a list of ServiceBuilders to a hash of output-format-specific definitions. It is practically the rendering pipeline for the DSL definitions
- * @param options
+ * Asynchronously generates output from a list of ServiceBuilders.
+ *
+ * This function is a key part of the rendering pipeline for DSL definitions. It takes a list of ServiceBuilders and converts them into a hash of output-format-specific definitions.
+ *
+ * @param {ReferenceResolver} params.runtime - The runtime reference resolver.
+ * @param {ServiceBuilder<any>[] | ServiceBuilder<any>} params.services - The list of ServiceBuilders to be converted.
+ * @param {OutputFormat<T>} params.outputFormat - The output format for the conversion (e.g. `HelmOutput` or `LocalrunOutput`).
+ * @param {EnvironmentConfig} params.env - The environment configuration.
+ *
+ * @returns {Promise<Services<T>>} A promise that resolves to a hash of output-format-specific definitions.
+ *
+ * @template T - The type of the service output.
  */
-export const generateOutput = async <T extends ServiceOutputType>(options: {
+export async function generateOutput<T extends ServiceOutputType>({
```

Comment:
Is all the refactoring from es6 arrow functions to regular functions required for the PR changes to work correctly?

Reply:
No, but the IDE help becomes clearer; this is a _function_ that does stuff, not a _variable_ storing information.

Reply:
So, just personal preference, I guess.

Reply:
Since it's not dependent on code behavior, what do you think about keeping the es6 syntax as is and we can take a team discussion about es6 vs regular functions in general?

I see no point refactoring these functions from es6 and not elsewhere.

Reply:
Sure, I'll change it back.

Reply:
Never mind, too much effort, and inconsistent already, anyways.

Reply:
we should be consistent…

Reply:
Agree. But there isn't clear technical direction (cc/ @saevarma). If we choose one or the other, it should be in another PR.

Reply:
We will schedule this topic for the next dev/devops discipline meeting.